src/jit/codegencommon.cpp

```diff
@@ -2465,6 +2465,21 @@ bool CodeGen::genCreateAddrMode(GenTreePtr  addr,
         return false;
     }
 
+#if defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND)
+    if (mode == -1)
+    {
+        if (rv1 != nullptr && rv2 == nullptr && cns == 0
+#if SCALED_ADDR_MODES
+            && mul == 0
+#endif
+            )
+        {
+            // we can just remove [rv1 + 0]
+            return false;
+        }
+    }
+#endif // defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND)
+
```

Comment:
I'll try for other targets too. But if it  will work OK propper place will be at:
https://github.com/dotnet/coreclr/blob/df7b49928acfd09fab63bbefbef8a8a1b325673b/src/jit/lower.cpp#L3752